### ISSUE TITLE
请问为什么tr和tcl都用两通道表示？各用一通道表示不就可以了吗

### ISSUE THREAD

**Issue by ustczhouyu** [princewang1994/TextSnake.pytorch#38]
请问为什么tr和tcl都用两通道表示？各用一通道表示不就可以了吗

**Comment by princevandev** [comment]
具体参考TextSnake原文的描述，公式（4）底下有一段话，TCL和TR使用4通道表示，目的是用softmax而不是softmax，个人觉得差别不大

**Comment by CMM915915** [comment]
请问画圆的时候的半径是怎么确定的啊？

**Comment by princevandev** [comment]
@CMM915915 不好意思这么久才回复，训练时半径如文中所述，用snake的宽度的一半画圆，测试时用预测的半径画圆

**Comment by chunchet-ng** [comment]
> 具体参考TextSnake原文的描述，公式（4）底下有一段话，TCL和TR使用4通道表示，目的是用softmax而不是softmax，个人觉得差别不大

Hi 您好，我不明白您指的“目的是用softmax而不是softmax”。有几个问题希望您能指点：
1）其实TCL和TR在原文里为什么需要各用两个通道呢？
2）两个通道之间的差别是什么？

谢谢

**Comment by princevandev** [comment]
@chunchet-ng TCL两通道+TR两通道一共4个通道，在每2个通道都使用softmax而不是使用sigmoid映射到0~1范围内，其实就是多分类和二分类的区别，这里用多分类的思想做二分类。使用单通道也可以，那么最后就需要用sigmoid函数了。

- 单通道的意思：这个像素是TCL的概率是0.9
- 两通道的意思：这个像素是TCL的概率是0.9，不是TCL的概率是0.1，softmax会让这两个通道的和为1，所以取一个argmax就知道这个像素是否被预测为TCL

如果仍然不明白可以去了解一下softmax分类，希望这个回答能帮到你。

**Comment by chunchet-ng** [comment]
谢谢您的回答，我明白了。

**Comment by xray1111** [comment]
> 具体参考TextSnake原文的描述，公式（4）底下有一段话，TCL和TR使用4通道表示，目的是用softmax而不是softmax，个人觉得差别不大

Do you mean using softmax instead of sigmoid?

**Comment by yuni1314** [comment]
好的
